### DISCUSSION TITLE
The Python Console drops errors

### DISCUSSION THREAD

**Discussion by Yazan-Zayn** [miguelgrinberg/Flask-SocketIO#1901] [Q&A] [answered]
Hello everyone
I have a raspberry pi and want to use Flask_SocketIO to control GPIOs.
I have the problem that the console drops errors.
  My Python code:

Python Console Errors :
![image](https://user-images.githubusercontent.com/74987943/200132773-ff15f827-f822-4aa0-a957-d1af04217482.png)

Browser Errors:
![image](https://user-images.githubusercontent.com/74987943/200132745-15b38cda-da0a-4a70-916a-2d16952e9e6f.png)

    Python Packages:
    
    eventlet==0.33.1
    ExplorerHAT==0.4.2
    Flask==2.1.3
    Flask-Cors==3.0.10
    Flask-InfluxDB==0.3.3
    Flask-Login==0.5.0
    Flask-Mail==0.9.1
    Flask-MySQLdb==0.2.0
    Flask-SocketIO==5.2.0
    Flask-SQLAlchemy==2.4.4
    python-decouple==3.1
    python-engineio==4.3.2
    python-prctl==1.7
    python-socketio==5.6.0
    Werkzeug==2.1.2
    wrapt==1.12.1
    zipp==3.9.0

Many thanks in advance

**Comment by miguelgrinberg** [comment, marked as answer]
I would start by uninstalling eventlet. Why are you using it here? This is an asynchronous framework, I don't think the GPIO library is compatible with it.

**Comment by Yazan-Zayn** [comment]
Thank you for your quick reply.

I know it works without eventlet but I am using eventlet for another script.
I didn't want to reinstall eventlet every time.

### For all others:
The solution is to uninstall eventlet.

**Comment by Yazan-Zayn** [comment]
I uninstalled eventlet and it works. 

After a few minutes the error message appears in the browser:



![image](https://user-images.githubusercontent.com/74987943/200137794-dbd5c1f8-ef49-4e65-90b3-0f1929b431f8.png)

**Reply by miguelgrinberg** [reply]
Install `simple-websocket` if you want WebSocket support.
